monai/transforms/spatial/array.py

```diff
@@ -3277,8 +3277,10 @@ def filter_count(self, image_np: NdarrayOrTensor, locations: np.ndarray) -> tupl
                 idx = argsort(image_np.sum(tuple(range(1, n_dims))))
             elif self.sort_fn == GridPatchSort.MAX:
                 idx = argsort(-image_np.sum(tuple(range(1, n_dims))))
+            elif self.sort_fn == GridPatchSort.RANDOM:
+                idx = np.random.permutation(image_np.shape[0])
```

Comment:
thanks for the PR, this randomised operation should be part of RandGridPatch and use self.R, similar to https://github.com/Project-MONAI/MONAI/blob/6f8e63f6cb2932492610515cd9b62f581fcb6d3e/monai/transforms/spatial/array.py#L3437
to ensure the pseudo random states are reproducible in different use cases. But I'm not sure how to implement it with minor changes, we may need to refactor both classes for this feature.